core/src/main/java/org/elasticsearch/gateway/GatewayMetaState.java

```diff
@@ -252,6 +257,23 @@ private void ensureNoPre019ShardState(NodeEnvironment nodeEnv) throws Exception
         }
     }
 
+    private void upgradePluginsMetaData() throws Exception {
+        MetaData metaData = loadMetaState();
+        boolean changed = false;
+        for (Function<MetaData, MetaData> metaDataUpgrader : pluginsService.getMetaDataUpgrader()) {
+            MetaData upgradedMetaData = metaDataUpgrader.apply(metaData);
```

Comment:
Please no `null` for no change needed, returning `Function.identity` is clear, and there is no need to make an optimization check.